gogs: fix directories ownership (runit service)

zeechs opened the pull request:
With the current `run` file, `gogs` is unable to start because its home directory is owned by `root`. It should be owned by the user `gogs` as the program needs to create a `.gitconfig` file in there.

Gottox commented:
Please increase the revision of the package :+1:

Duncaen commented:
I don't think its a good idea to recursively chown the home directory each time the service starts.

zeechs commented:
True that if the `run` script is launched by a regular user, their whole home directory would be messed up.

I reverted my changes and just added the originally missing chown (for `${HOME}` and `${HOME}/data`).
Now what if the user already has a `data` or `repo` folder in their home directory? Should this case be considered?

I also increased the revision number in the template.

Duncaen commented:
But still, why would the home directory owned by root, shouldn't this be fixed at the root and not while starting the service?

zeechs commented:
- moved `/etc/gogs.ini` to `/etc/gogs/conf.ini`.
On first start, Gogs attempts to create a temporary file in `/etc/` (beside its .ini file) but fails with the following error:
`Fail to save configuration: open /etc/gogs.ini.139946627.tmp: permission denied`
Having the .ini file in a subdir of `/etc/` with write permissions for user gogs fixed the issue.
I suppose this change would break the installation in case a user is updating from the previous revision of the package.

- create directories using `make_dirs` rather than doing it in the `run` script.
note: subdirectories of gogs' home are created by Gogs if they don't exist, so the package doesn't need to create them.

- cleaned up run script and start the service as user gogs.


There is one more potential issue:
gogs modifies the /etc/gogs/conf.ini file
this triggers an error when uninstalling:

```
$ sudo xbps-remove -R gogs

Name Action    Version           New version            Download size
gogs remove    0.11.43_2         -                      -

Size freed on disk:             41MB
Space available on disk:       192GB

Do you want to continue? [Y/n]
Removing `gogs-0.11.43_2' ...
Disabled gogs (gogs) system user/group.
ERROR: gogs-0.11.43_2: configuration file `/etc/gogs/conf.ini' SHA256 mismatch, preserving file
Removed `gogs-0.11.43_2' successfully.

0 downloaded, 0 installed, 0 updated, 0 configured, 1 removed.
```

Is it supposed to stay this way ?

Gottox commented on srcpkgs/gogs/INSTALL:
```diff
@@ -3,7 +3,7 @@
 post)
 	if [ "$UPDATE" = "no" ]; then
-		chown gogs:gogs etc/gogs.ini
+		chown gogs:gogs /etc/gogs/conf.ini
```
Paths in INSTALL scripts must be relative in case xbps-install is used with `--rootdir <PATH>` and `chroot(2)` isn't available.

zeechs replied:
Done!

maxice8 commented:
ping
